programs/stake_api/src/stake_state.rs

```diff
@@ -276,6 +282,33 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
             Err(InstructionError::InvalidAccountData)
         }
     }
+    fn withdraw(
+        &mut self,
+        lamports: u64,
+        to: &mut KeyedAccount,
+        current: &syscall::current::Current,
+    ) -> Result<(), InstructionError> {
+        if self.signer_key().is_none() {
+            return Err(InstructionError::MissingRequiredSignature);
+        }
+
+        if let StakeState::Stake(stake) = self.state()? {
+            // Assume full stake if the stake is under warmup/cooldown
+            let staked = if stake.stake(current.epoch) == 0 {
+                0
+            } else {
+                stake.stake
+            };
+            if lamports > self.account.lamports.saturating_sub(staked) {
+                return Err(InstructionError::InsufficientFunds);
+            }
```

Comment:
I think we should also lower stake.stake to whatever remains in the account

Reply:
: